extensions/ql-vscode/src/remote-queries/remote-queries-manager.ts

```diff
@@ -119,21 +119,47 @@ export class RemoteQueriesManager extends DisposableObject {
     uri: Uri | undefined,
     progress: ProgressCallback,
     token: CancellationToken
-  ): Promise<void> {
+  ): Promise<RemoteQuery | undefined> {
     const credentials = await Credentials.initialize(this.ctx);
 
-    const querySubmission = await runRemoteQuery(this.cliServer, credentials, uri || window.activeTextEditor?.document.uri, progress, token, this.variantAnalysisManager);
+    const {
+      actionBranch,
+      base64Pack,
+      repoSelection,
+      queryFile,
+      queryMetadata,
+      controllerRepo,
+      queryStartTime,
+      language,
+    } = await prepareRemoteQueryRun(this.cliServer, credentials, uri, progress, token);
+
+    const apiResponse = await runRemoteQueriesApiRequest(credentials, actionBranch, language, repoSelection, controllerRepo, base64Pack);
+
+    if (!apiResponse) {
+      return undefined;
+    }
 
-    if (querySubmission?.query) {
-      const query = querySubmission.query;
-      const queryId = this.createQueryId();
+    const workflowRunId = apiResponse.workflow_run_id;
+    const repositoryCount = apiResponse.repositories_queried.length;
+    const query = await buildRemoteQueryEntity(
+      queryFile,
+      queryMetadata,
+      controllerRepo,
+      queryStartTime,
+      workflowRunId,
+      language,
+      repositoryCount
+    );
 
-      await this.prepareStorageDirectory(queryId);
-      await this.storeJsonFile(queryId, 'query.json', query);
+    const queryId = this.createQueryId();
 
-      this.remoteQueryAddedEventEmitter.fire({ queryId, query });
-      void commands.executeCommand('codeQL.monitorRemoteQuery', queryId, query);
-    }
+    await this.prepareStorageDirectory(queryId);
+    await this.storeJsonFile(queryId, 'query.json', query);
+
+    this.remoteQueryAddedEventEmitter.fire({ queryId, query });
+    void commands.executeCommand('codeQL.monitorRemoteQuery', queryId, query);
+
+    return query;
```

Comment:
Seems like this is only used in tests - is that correct? I think we should generally avoid having some code just for our tests but appreciate that would probably require a bunch more changes to avoid it.

Reply:
Thanks for mentioning this! I completely agree that we should not be doing this and I don't think it'll be a lot of work to remove (I can instead check that the `monitorRemoteQuery` command has been called), so I'll remove it now rather than leaving it in.